### PR TITLE
SitesOverview: split utils functions into separate files

### PR DESCRIPTION
**Pull request by mattgawarecki**
This pull request extracts all the functions and constants inside `client/jetpack-cloud/sections/agency-dashboard/sites-overview/utils.ts` into separate files, and organizes them so that they're located nearer to where they're used.

The code itself is unchanged by-and-large, but some refactoring was necessary to mitigate the use of the raw `translate` function and replace it with `useTranslate`, as recommended in the `i18n-calypso` package documentation.

Resolves https://github.com/Automattic/jetpack-genesis/issues/16.

## Proposed Changes

* Extract functions and constants from `utils.ts` into various new files. Generally, `lib` directories are for non-hook helper functions and `hooks` are for React hooks.
* Replace all references to `utils.ts` with updated references to the newly-relocated code.
* Migrate and update all unit test cases for existing functions.
* Remove `utils.ts`.

## Testing Instructions

* Examine the code changes in this PR commit-by-commit, verifying that the changes at each step make sense.
* Ensure all unit and integration tests pass.
* Ensure all type checks pass.
* Fully test all features on the main Pro Dashboard page to ensure nothing has changed and no new errors show up. Thoroughly test the sites table, bulk management interface, and settings for each product (e.g., Backup, Scan, Monitor).

### CODE REVIEW

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-card/test/site-card.tsx [hunk 1]
```diff
@@ -103,7 +102,7 @@ describe( '<SiteCard>', () => {
 		const { container } = render(
 			<Provider store={ store }>
 				<QueryClientProvider client={ queryClient }>
-					<SiteCard rows={ rows } columns={ siteColumns } />
+					<SiteCard rows={ rows } columns={ [] } />
```
Comment (mattgawarecki) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-card/test/site-card.tsx:105]: This test doesn't require real column data to pass.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-table-row/test/site-table-row.tsx [hunk 2]
```diff
@@ -112,7 +111,7 @@ describe( '<SiteTableRow>', () => {
 	};
 	const props = {
 		item,
-		columns: siteColumns,
+		columns: [],
```
Comment (mattgawarecki) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-table-row/test/site-table-row.tsx:114]: This test doesn't require real column data to pass.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-table/test/site-table.tsx [hunk 3]
```diff
@@ -117,7 +116,41 @@ describe( '<SiteTable>', () => {
...
+				title: 'Monitor',
+				className: 'min-width-100px',
+				isExpandable: true,
+			},
+			{
+				key: 'plugin',
+				title: 'Plugins',
+				className: 'width-fit-content',
+			},
+		],
```
Comment (mattgawarecki) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-table/test/site-table.tsx:154]: This test requires column data to pass, but we can pass in static values.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/test/utils.ts
Comment (mattgawarecki) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/test/utils.ts:1]: All the tests in this file were preserved and migrated to separate files appropriate to the function they were testing.

**Review by mattgawarecki — COMMENTED** [review]
A few comments for reviewers -->

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/lib/constants.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Returns the product slug that can be purchased from the dashboard.
```
Comment (jkguidaven) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/lib/constants.ts]: We probably want to change the text here

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/hooks/use-extracted-backup-title.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
+import { useTranslate } from 'i18n-calypso';
+import { useMemo } from 'react';
+import { Backup } from '../../types';
+
+export function extractBackupTextValues( str: string ): { [ key: string ]: number } {
```
Comment (jkguidaven) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/hooks/use-extracted-backup-title.ts]: Just nitpicking, we can use arrow function here for consistency.

**Comment by jkguidaven** [comment]
> Generally, lib directories are for non-hook helper functions, and hooks are for React hooks.

Hi, @mattgawarecki. Can you enlighten me on how we differentiate a piece of code as a `util` or `lib`? I see both folder names being used in Calypso, but I don't have a clear understanding of how it differs.

**Review by jkguidaven — APPROVED** [review]
Excellent work on refactoring the util functions and fixing the use of the raw `translate` function. The code LGTM and I don't have any blocking comments. 👍 

Did a test on the main Pro Dashboard functionality and didn't find any issues. 

I also would suggest another pair of eyes to look at this to ensure I didn't miss anything.

**Comment by mattgawarecki** [comment]
> Hi, @mattgawarecki. Can you enlighten me on how we differentiate a piece of code as a `util` or `lib`? I see both folder names being used in Calypso, but I don't have a clear understanding of how it differs.

I would say that generally, `util` and `lib` are close enough in meaning that they're basically interchangeable, at least in their stated purpose. I leaned more toward `lib` in this refactor, because I have a hunch that it's _maybe_ (?) a little less likely people will store things here that aren't used in at least a few places. I get the impression that `lib` implies "these things are meant to be re-used, and they're used in several places," whereas `util` is more relaxed/less organized. TL;DR it's probably a subjective thing. 🙂

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8808479

Hi @mattgawarecki, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `3 minutes`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8808479).

**Pull request merged**